### PR TITLE
Separate password update form for manage settings

### PR DESCRIPTION
**Pull request by ramillim**
@williamcodes 
This PR addresses the item for having a separate form for changing the password: https://github.com/heatseeknyc/heatseeknyc/issues/123

The open PR https://github.com/heatseeknyc/heatseeknyc/pull/135 should be merged first so that all the css changes don't show up on this diff.

This implements solution 3 in the devise wiki: https://github.com/plataformatec/devise/wiki/How-To:-Allow-users-to-edit-their-password

image

image

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -52,6 +52,21 @@ def show
     end
   end
 
+  def edit_password
+    @user = current_user
+  end
+
+  def update_password
+    @user = User.find(current_user.id)
```
Comment (williamcodes) [inline app/controllers/users_controller.rb]: Why not use current_user as-is?

```
@user = current_user
```

File: app/helpers/user_controller_helper.rb [hunk 2]
```diff
@@ -6,4 +6,16 @@ def render_collaborators_for(user)
...
+    :user
+  end
+
+  def resource
+    @user
+  end
+
+  def devise_mapping
+    @devise_mapping ||= Devise.mappings[:user]
+  end
```
Comment (williamcodes) [inline app/helpers/user_controller_helper.rb]: Is the `resource_name` method used anywhere? Also the `resource` method, I don't see those being used. Is this devise magic? Must be devise magic. No other reason to cache a `@devise_mapping` instance variable.

**Comment by williamcodes** [comment]
Brilliant! Love it, particularly the test coverage.

**Comment by williamcodes** [comment]
Normally I would let you do the rebase since you're more qualified having written the code and so the PR stays intact, but since you're deployed at the moment I've gone ahead and done it in a separate PR: #143

**Comment by williamcodes** [comment]
closed by #143

File: app/helpers/user_controller_helper.rb (hunk 2 above)
Comment (ramillim) [inline app/helpers/user_controller_helper.rb]: It was a Devise thing that threw a method_missing error unless it was defined. This was the example in the Devise wiki:
https://github.com/plataformatec/devise/wiki/How-To:-Display-a-custom-sign_in-form-anywhere-in-your-app

File: app/controllers/users_controller.rb (hunk 1 above)
Comment (ramillim) [inline app/controllers/users_controller.rb]: I believe I was over thinking this when I was trying to guard against a user trying to post request to another user id's password. You're totally correct, this should just be current_user.

**Pull request closed without merging**